Solve new mypy issues

```sh
src/xtgeo/grid3d/_gridprop_export.py:110: error: Incompatible types in assignment (expression has type "ndarray[Any, Any]", variable has type "MaskedArray[tuple[int], dtype[Any]]")  [assignment]
src/xtgeo/grid3d/grid_property.py:525: error: Incompatible return value type (got "dtype[Any]", expected "type[unsignedinteger[_8Bit]] | type[unsignedinteger[_16Bit]] | type[signedinteger[_16Bit]] | type[signedinteger[_32Bit]] | type[signedinteger[_64Bit]] | type[floating[_16Bit]] | type[floating[_32Bit]] | type[float64]")  [return-value]
src/xtgeo/grid3d/grid_property.py:972: error: Argument "fill_value" to "filled" has incompatible type "Buffer | _SupportsArray[dtype[Any]] | _NestedSequence[_SupportsArray[dtype[Any]]] | complex | bytes | str | _NestedSequence[complex | bytes | str]"; expected "complex | str | bytes | generic[Any] | None"  [arg-type]
src/xtgeo/grid3d/grid_property.py:1222: error: Incompatible return value type (got "ndarray[Any, Any]", expected "MaskedArray[tuple[Any, ...], dtype[Any]] | None")  [return-value]
src/xtgeo/grid3d/_grid_etc1.py:425: error: Incompatible types in assignment (expression has type "ndarray[tuple[int], dtype[signedinteger[_32Bit | _64Bit]]]", variable has type "signedinteger[_32Bit | _64Bit]")  [assignment]
src/xtgeo/grid3d/_grid_etc1.py:426: error: Incompatible types in assignment (expression has type "ndarray[tuple[int], dtype[signedinteger[_32Bit | _64Bit]]]", variable has type "signedinteger[_32Bit | _64Bit]")  [assignment]
src/xtgeo/grid3d/_grid_etc1.py:427: error: Incompatible types in assignment (expression has type "ndarray[tuple[int], dtype[signedinteger[_32Bit | _64Bit]]]", variable has type "signedinteger[_32Bit | _64Bit]")  [assignment]
src/xtgeo/grid3d/_grid_etc1.py:441: error: Incompatible types in assignment (expression has type "GridProperty", variable has type "signedinteger[_32Bit | _64Bit]")  [assignment]
src/xtgeo/grid3d/_grid_etc1.py:449: error: Incompatible types in assignment (expression has type "GridProperty", variable has type "signedinteger[_32Bit | _64Bit]")  [assignment]
src/xtgeo/grid3d/_grid_etc1.py:457: error: Incompatible types in assignment (expression has type "GridProperty", variable has type "signedinteger[_32Bit | _64Bit]")  [assignment]
src/xtgeo/grid3d/_grid_etc1.py:467: error: Incompatible return value type (got "tuple[signedinteger[_32Bit | _64Bit], signedinteger[_32Bit | _64Bit], signedinteger[_32Bit | _64Bit]]", expected "tuple[GridProperty, GridProperty, GridProperty]")  [return-value]
```

--- a/src/xtgeo/grid3d/_grid_etc1.py
+++ b/src/xtgeo/grid3d/_grid_etc1.py
@@ -420,11 +420,11 @@ def get_ijk(
     """Get I J K as properties."""
     ashape = self.dimensions
 
-    ix, jy, kz = np.indices(ashape)
+    ix_idx, jy_idx, kz_idx = np.indices(ashape)
 
-    ix = ix.ravel()
-    jy = jy.ravel()
-    kz = kz.ravel()
+    ix = ix_idx.ravel()
+    jy = jy_idx.ravel()
+    kz = kz_idx.ravel()
 
     if asmasked:
         actnum = self.get_actnum()
@@ -438,7 +438,7 @@ def get_ijk(
         jy += 1
         kz += 1
 
-    ix = GridProperty(
+    ix_gprop = GridProperty(
         ncol=self._ncol,
         nrow=self._nrow,
         nlay=self._nlay,
@@ -446,7 +446,7 @@ def get_ijk(
         name=names[0],
         discrete=True,
     )
-    jy = GridProperty(
+    jy_gprop = GridProperty(
         ncol=self._ncol,
         nrow=self._nrow,
         nlay=self._nlay,
@@ -454,7 +454,7 @@ def get_ijk(
         name=names[1],
         discrete=True,
     )
-    kz = GridProperty(
+    kz_gprop = GridProperty(
         ncol=self._ncol,
         nrow=self._nrow,
         nlay=self._nlay,
@@ -464,7 +464,7 @@ def get_ijk(
     )
 
     # return the objects
-    return ix, jy, kz
+    return ix_gprop, jy_gprop, kz_gprop
 
 
 def get_ijk_from_points(

--- a/src/xtgeo/grid3d/_gridprop_export.py
+++ b/src/xtgeo/grid3d/_gridprop_export.py
@@ -20,6 +20,8 @@ from xtgeo.io._file import FileFormat, FileWrapper
 from ._roff_parameter import RoffParameter
 
 if TYPE_CHECKING:
+    import numpy.typing as npt
+
     from xtgeo.common.types import FileLike
 
     from .grid_property import GridProperty
@@ -104,7 +106,7 @@ def _export_grdecl(
     fmt: str | None = None,
 ) -> None:
     """Export ascii or binary GRDECL"""
-    vals = gridprop.values.ravel(order="F")
+    vals: npt.NDArray = gridprop.values.ravel(order="F")
     if np.ma.isMaskedArray(vals):
         undef_export = 1 if gridprop.isdiscrete or "int" in str(dtype) else 0.0
         vals = np.ma.filled(vals, fill_value=undef_export)

--- a/src/xtgeo/grid3d/grid_property.py
+++ b/src/xtgeo/grid3d/grid_property.py
@@ -7,6 +7,7 @@ from types import FunctionType
 from typing import TYPE_CHECKING, Any, Literal
 
 import numpy as np
+import numpy.typing as npt
 
 import xtgeo
 from xtgeo.common import XTGeoDialog, null_logger
@@ -47,16 +48,6 @@ if TYPE_CHECKING:
     from ._gridprop_op1 import XYValueLists
     from .grid import Grid
 
-    GridProperty_DType = Union[
-        type[np.uint8],
-        type[np.uint16],
-        type[np.int16],
-        type[np.int32],
-        type[np.int64],
-        type[np.float16],
-        type[np.float32],
-        type[np.float64],
-    ]
     Roxar_DType = Union[type[np.uint8], type[np.uint16], type[np.float32]]
 
 # --------------------------------------------------------------------------------------
@@ -509,7 +500,7 @@ class GridProperty(_Grid3D):
             self.discrete_to_continuous()
 
     @property
-    def dtype(self) -> GridProperty_DType:
+    def dtype(self) -> npt.DTypeLike:
         """
         Get or set the ``values`` numpy dtype.
 
@@ -525,8 +516,8 @@ class GridProperty(_Grid3D):
         return self._values.dtype
 
     @dtype.setter
-    def dtype(self, dtype: GridProperty_DType) -> None:
-        allowed: list[GridProperty_DType] = (
+    def dtype(self, dtype: npt.DTypeLike) -> None:
+        allowed: list[npt.DTypeLike] = (
             [np.uint8, np.uint16, np.int16, np.int32, np.int64]
             if self.isdiscrete
             else [np.float16, np.float32, np.float64]
@@ -957,15 +948,17 @@ class GridProperty(_Grid3D):
             Non-masked array copy of 3D-shaped values
 
         """
+        fvalue: float | int
         if fill_value is None:
             if self._isdiscrete:
-                fvalue: npt.ArrayLike = UNDEF_INT
+                fvalue = UNDEF_INT
                 dtype: type[np.int32] | type[np.float64] = np.int32
             else:
                 fvalue = UNDEF
                 dtype = np.float64
         else:
-            fvalue = fill_value
+            # TODO: possible bug here
+            fvalue = fill_value  # type: ignore
             dtype = np.float64
 
         val = self.values.copy().astype(dtype)
@@ -1181,7 +1174,7 @@ class GridProperty(_Grid3D):
 
     def get_values_by_ijk(
         self, iarr: np.ndarray, jarr: np.ndarray, karr: np.ndarray, base: int = 1
-    ) -> np.ma.MaskedArray | None:
+    ) -> npt.NDArray | None:
         """
         Get a 1D ndarray of values by I J K arrays.
 

